Sources/xcresultparser/CoberturaCoverageConverter.swift

```diff
@@ -60,44 +111,55 @@ public class CoberturaCoverageConverter: CoverageConverter, XmlSerializable {
         let packagesElement = XMLElement(name: "packages")
         rootElement.addChild(packagesElement)
 
+        // Get the xccov results as a JSON.
+        var arguments = ["xccov", "view"]
+        if resultFile.url.pathExtension == "xcresult" {
+            arguments.append("--archive")
+        }
+        arguments.append("--json")
+        arguments.append(resultFile.url.path)
+        let coverageData = try Shell.execute(program: "/usr/bin/xcrun", with: arguments)
+        let resultsString = String(decoding: coverageData, as: UTF8.self)
+        guard let jsonData = resultsString.data(using: String.Encoding.utf8) else {
+            writeToStdError("Failed to convert to Data")
+            throw JSONParseError.convertError(code: 0, message: "Failed to convert to Data object")
+        }
+        guard let coverageJson = try? JSONDecoder().decode(FileCoverage.self, from: jsonData) else {
+            writeToStdError("Failed to convert to JSON")
+            throw JSONParseError.convertError(code: 0, message: "Failed to convert to JSON Object")
+        }
+
         let fileInfoSemaphore = DispatchSemaphore(value: 1)
         var fileInfo: [FileInfo] = []
-        
-        // since we need to invoke xccov for each file, it takes pretty much time
-        // so we invoke it in parallel on 8 threads, that speeds up things considerably
         let queue = OperationQueue()
```

Comment:
Now that xcccv tool doesn't need to get called repeatedly we do not need to spread the for loop over different threads. In fact I testest and without spawning to threads it ran slightly faster.

Reply:
Hey there, sorry for the delay here, haven't been at my computer much this weekend. Is there a way to toggle this to be a CLI flag? I imagine that for a very large results file it might be faster to work serially on a machine with not a lot of CPUs but on a machine with 8+ cpus I image it might eek out better performance? I can try and work on a PR to add that functionality if you'd be alright with that

Reply:
I was just saving the CPU for other parallel tasks. I will try some larger xcresult bundles and measure the difference with the PR I just created and with your version. Let's see, it it is worth it to block more than one core for this. And let's take this discussion to the new PR. I invited you as collaborator and once you accept I can add you as reviewer to the PR.

Reply:
Sounds good to me, I'm not all too familiar with Swift's parallelism and my best frame of reference is Go's goroutines (greenthreads)